src/store/reducers/TagsReducer.js

```diff
@@ -0,0 +1,42 @@
+import { FETCH_TAGS_ACTION } from '../actions/actionTypes';
+import {ENDED, FAILED, STARTED, SUCCEEDED} from "../actions/async";
+
+export const defaultState ={
+    tags: [],
+    status: "succeeded",
+    errors: ""
+};
+
+const tagsReducer = (state=defaultState, action) => {
+    if (typeof action === "undefined"){
```

Comment:
Just a quick question here. Doesn't the default return handle this in case the action is not defined?
I mean wouldn't having this if statement and the default return in the switch case be kind of repeating the same thing.

Reply:
@FahdJamy Thanks for that, I guess you're right. For anything other than the defined action-types, default state should be returned. thanks

Reply:
Nope, the default won't handle this, because we shall attempt to access `action.type` in the case, which will throw an error if `action` is `undefined`.
It is necessary IMO.